src/Shared/PackageManagers/CPANProjectManager.cs

```diff
@@ -120,7 +120,7 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
                 }
                 if (doExtract)
                 {
-                    downloadedPaths.Add(await ExtractArchive(targetName, await result.Content.ReadAsByteArrayAsync(), cached));
+                    downloadedPaths.Add(await ExtractArchive(TopLevelExtractionDirectory, targetName, await result.Content.ReadAsByteArrayAsync(), cached));
```

Comment:
Why are we converting the Stream to an array? I think this work if you just pass the stream instead of reading hteb ytes and then making a new stream to hold it.

Reply:
If that has not been working with RecursiveExtractor let me know and I can investigate.

Reply:
I think it works fine now.